src/core/metastep.h

```diff
@@ -138,13 +140,15 @@ class OOFEM_EXPORT MetaStep
     int giveLastStepNumber() { return ( sindex + numberOfSteps - 1 ); }
     /// Returns class name of receiver.
     const char *giveClassName() const { return "MetaStep"; }
-
+    /// Returns true if last step
+    bool isLastStep(TimeStep *tStep) { return ( tStep->giveTargetTime() == finalTime); }
```

Comment:
The number might have numerical artifacts and the equality will return false even for the last step.